R/update-news.R

```diff
@@ -348,6 +394,30 @@ check_gh_pat <- function() {
       )
     )
   }
+
+  # check scopes for PR
+  v4_scopes <- c(
+    "repo", "read:packages",
+    "read:org", "read:public_key", "read:repo_hook",
+    "user", "read:discussion", "read:enterprise",
+    "read:gpg_key"
+    )
+  scopes <- if (nzchar(Sys.getenv("FLEDGE_TEST_SCOPES"))) {
+    v4_scopes
+  } else {
+    trimws(strsplit(gh::gh_whoami()[["scopes"]], ",")[[1]])
+  }
+
+
+  missing_scopes <- v4_scopes[!(v4_scopes %in% scopes)]
+  if (length(missing_scopes) > 0) {
+        rlang::warn(
+      message = c(
+        x = sprintf("Missing scopes for GitHub GraphQL API (used for finding issues linked to PR): %s", toString(missing_scopes)),
```

Comment:
still not sure whether we should warn here

Reply:
(as I'm unsure which missing scope is really an issue, it's not documented, it seems over the top to try them all)

Reply:
@krlmlr opinion?

Reply:
Is this the right place? We give a hard error elsewhere if scopes are missing. (I typically set `GITHUB_PAT` manually in every session I need it.)

Reply:
The check that leads to an error is done in the same function just before this. There is also `check_gh_scopes()` that is used in code that really needs the repo scope for opening PRs.

The thing is, I'm not sure not having the right scopes warrants an error especially as the fledge user might not even use the PR message feature.

`check_gh_scopes()` and `check_gh_pat()` could be merged and have an argument that'd define what scope is checked and whether it's a show stopper, however before doing that I'd like to confirm it's ok to **not** error if there are not the V4 scopes (which only means we can't "include reference to issue closed with a PR").

Reply:
Failing hard is okay for now. Once we have a way to add this information later on, we can turn this into a message (and/or an entry in `NEWS.md`) with corresponding instructions.